library/src/main/java/com/novoda/downloadmanager/RoomAppDatabase.java

```diff
@@ -8,24 +8,33 @@
 @Database(entities = {RoomBatch.class, RoomFile.class}, version = 1)
 abstract class RoomAppDatabase extends RoomDatabase {
 
-    private static volatile RoomAppDatabase SINGLE_INSTANCE;
+    private static volatile RoomAppDatabase singleInstance;
+    private static final Object LOCK = new Object();
 
     abstract RoomBatchDao roomBatchDao();
 
     abstract RoomFileDao roomFileDao();
 
     static RoomAppDatabase obtainInstance(Context context) {
-        if (SINGLE_INSTANCE == null) {
-            SINGLE_INSTANCE = newInstance(context);
+        RoomAppDatabase database = singleInstance;
```

Comment:
This post gives a rather nice explanation for thread safe singletons 😄  https://www.journaldev.com/1377/java-singleton-design-pattern-best-practices-examples#thread-safe-singleton

Reply:
Yup